include/zephyr/drivers/i2c.h

```diff
@@ -994,7 +1005,12 @@ static inline void i2c_iodev_submit(struct rtio_iodev_sqe *iodev_sqe)
 	const struct device *dev = dt_spec->bus;
 	const struct i2c_driver_api *api = (const struct i2c_driver_api *)dev->api;
 
-	api->iodev_submit(dt_spec->bus, iodev_sqe);
+	if (api->iodev_submit != NULL) {
```

Comment:
IMHO this is the wrong place to do this, instead we should do it when initializing the function pointer tables and avoid this branch on every call to submit.